WcaOnRails/spec/controllers/teams_controller_spec.rb

```diff
@@ -65,138 +47,128 @@
   describe 'POST #create' do
     context 'when not signed in' do
       it 'redirects to the sign in page' do
-        post :create, team: { name: "Team2016" }
+        post :create, committee_id: team.committee.slug, team: {name: "Team 2016", description: "An important team for 2016"}
         expect(response).to redirect_to new_user_session_path
       end
     end
 
     context 'when signed in as a regular user' do
       sign_in { FactoryGirl.create :user }
+
       it 'does not allow creation' do
-        post :create, team: { name: "Team2016" }
+        post :create, committee_id: team.committee.slug, team: {name: "Team 2016", description: "An important team for 2016"}
         expect(response).to redirect_to root_url
       end
     end
 
     context 'when signed in as an admin' do
-      sign_in { FactoryGirl.create :admin }
+      sign_in { FactoryGirl.create(:admin) }
 
       it 'creates a new team' do
-        post :create, team: { name: "Team2016" }
-        new_team = Team.find_by_name("Team2016")
-        expect(response).to redirect_to edit_team_path(new_team)
-        expect(new_team.name).to eq "Team2016"
+        post :create, committee_id: team.committee.slug, team: {name: "Team 2016", description: "An important team for 2016"}
+        new_team = Team.find_by_name("Team 2016")
+        expect(response).to redirect_to committee_path(team.committee.slug)
+        expect(new_team.name).to eq "Team 2016"
+      end
+    end
+
+    context 'when signed in as a demoted admin' do
+      sign_in { FactoryGirl.create :admin_demoted }
+
+      it 'does not allow creation' do
+        post :create, committee_id: team.committee.slug, team: {name: "Team 2016", description: "An important team for 2016"}
+        expect(response).to redirect_to root_url
       end
     end
   end
 
   describe 'GET #edit' do
     context 'when signed in as a team leader without rights to manage all teams' do
-      let(:team_where_is_leader) { Team.find_by_friendly_id('wrc') }
-      let(:team_where_is_not_leader) { Team.find_by_friendly_id('software') }
-      let(:leader) do
-        user = FactoryGirl.create(:user)
-        FactoryGirl.create(:team_member, team_id: team_where_is_leader.id, user_id: user.id, team_leader: true)
-        user
-      end
+      let(:leader) { FactoryGirl.create(:regulations_team_leader) }
+      let(:team_where_is_leader) { Team.find_by_slug('regulations-team') || FactoryGirl.create(:team, name: 'Regulations Team', committee: Committee.find_by_slug(Committee::WCA_REGULATIONS_COMMITTEE)) }
+      let(:team_where_is_not_leader) { Team.find_by_slug('software-team') || FactoryGirl.create(:team, name: 'Software Team', committee: Committee.find_by_slug(Committee::WCA_SOFTWARE_COMMITTEE)) }
```

Comment:
1. You've been using constants in `Team.find_by_slug(...)` so far.
2. This `find || create` pattern is ugly, as a lot of other code relies on the current teams maybe it's worth seeding them the same way we do with countries.